packages/flutter/lib/src/widgets/_window_positioner.dart

```diff
@@ -0,0 +1,546 @@
+            parentRect,
+            parentAnchor.flipY().flipX().anchorPositionFor(anchorRect) + _flipX(_flipY(offset)),
+          ) +
+          childAnchor.flipY().flipX().offsetFor(childSize);
+      if (_rectContains(outputRect, result & childSize)) {
+        return result & childSize;
+      }
+    }
+
+    {
+      Offset result =
+          _constraintTo(parentRect, parentAnchor.anchorPositionFor(anchorRect) + offset) +
+          childAnchor.offsetFor(childSize);
+
+      if (constraintAdjustment.contains(WindowPositionerConstraintAdjustment.slideX)) {
+        final double leftOverhang = result.dx - outputRect.left;
+        final double rightOverhang = result.dx + childSize.width - outputRect.right;
+        if (leftOverhang < 0.0) {
+          result = result.translate(-leftOverhang, 0.0);
+        } else if (rightOverhang > 0.0) {
+          result = result.translate(-rightOverhang, 0.0);
+        }
+      }
+
+      if (constraintAdjustment.contains(WindowPositionerConstraintAdjustment.slideY)) {
+        final double topOverhang = result.dy - outputRect.top;
+        final double bottomOverhang = result.dy + childSize.height - outputRect.bottom;
+        if (topOverhang < 0.0) {
+          result = result.translate(0.0, -topOverhang);
+        } else if (bottomOverhang > 0.0) {
+          result = result.translate(0.0, -bottomOverhang);
+        }
+      }
+
+      if (_rectContains(outputRect, result & childSize)) {
+        return result & childSize;
+      }
+    }
+
+    {
+      Offset result =
+          _constraintTo(parentRect, parentAnchor.anchorPositionFor(anchorRect) + offset) +
+          childAnchor.offsetFor(childSize);
+
+      if (constraintAdjustment.contains(WindowPositionerConstraintAdjustment.resizeX)) {
+        final double leftOverhang = result.dx - outputRect.left;
+        final double rightOverhang = result.dx + childSize.width - outputRect.right;
+        if (leftOverhang < 0.0) {
+          result = result.translate(-leftOverhang, 0.0);
+          childSize = Size(childSize.width + leftOverhang, childSize.height);
+        }
+        if (rightOverhang > 0.0) {
+          childSize = Size(childSize.width - rightOverhang, childSize.height);
+        }
+      }
+
+      if (constraintAdjustment.contains(WindowPositionerConstraintAdjustment.resizeY)) {
+        final double topOverhang = result.dy - outputRect.top;
+        final double bottomOverhang = result.dy + childSize.height - outputRect.bottom;
+        if (topOverhang < 0.0) {
+          result = result.translate(0.0, -topOverhang);
+          childSize = Size(childSize.width, childSize.height + topOverhang);
+        }
+        if (bottomOverhang > 0.0) {
+          childSize = Size(childSize.width, childSize.height - bottomOverhang);
+        }
+      }
+
+      if (_rectContains(outputRect, result & childSize)) {
+        return result & childSize;
+      }
+    }
+
+    return defaultResult;
+  }
+}
+
+extension on WindowPositionerAnchor {
```

Comment:
If we keep the `WindowPositionerAnchor` enum, could we move these to be methods on the enum? See: https://dart.dev/language/enums#declaring-enhanced-enums